packages/fast-components/src/badge/badge.template.ts

```diff
@@ -0,0 +1,14 @@
+import { html } from "@microsoft/fast-element";
+import { Badge } from "./badge";
+
+export const BadgeTemplate = html<Badge>`
+    <template>
+        <div
+            class="badge"
+            style="background-color: var(--badge-fill-${x =>
+                x.fill}); color: var(--badge-color-${x => x.color})"
+        >
+            <slot></slot>
+        </div>
```

Comment:
It is because we can't assign styling inline in the template and pass styling to the component implementation. For example:

```
<fast-badge style="font-size: 20px"/>
```

Reply:
@eljefe223 @EisenbergEffect are we not able to check for and merge the style object within the model itself? Setting the fill and color seems to be the primary purpose of the markup here.

Reply:
We aren't at least based on conversations I had with @nicholasrice . I tried a couple different methods none seemed to work. Would be a good feature request for `fast-element` it is possible.

Reply:
Currently, we can't leverage host directives for `style` or `class` because it would clobber anything the customer put on the host themself. So, either we need a wrapper that we can bind to, or we need to execute code to manually add/remove classes and styles from the host.

It's possible we'll be able to have smarter class and style binding in the future. I've been thinking about it, but it hasn't bubbled up to the top of my work list yet.

Reply:
Note: I think we can definitely have smarter class binding and possibly have smarter style binding. The style part is harder. I think I've got some prior art lying around but haven't spent the time to research all that yet.